DataFormats/Reconstruction/include/ReconstructionDataFormats/TrackTPCITS.h

```diff
@@ -64,14 +65,14 @@ class TrackTPCITS : public o2::track::TrackParCov
   void print() const;
 
  private:
-  int mRefTPC = -1;                  ///< reference on ITS track entry in its original container
-  int mRefITS = -1;                  ///< reference on TPC track entry in its original container
+  GlobalTrackID mRefTPC;             ///< reference on ITS track entry in its original container
+  GlobalTrackID mRefITS;             ///< reference on TPC track entry in its original container
```

Comment:
I am not fully sure if it is better to have the GlobalTrackID here directly, or keep the mRefTPC and mRefITS as integer as before, since it is clear where they belong. The conversion to the globalTrackId could also happen in the GlobalTrackID getRefTPC() accessor (which would be just one or operation), there could even be 2 accessors, returning the int for users that know the container, and returning the global ID for general purposes. But OK, not sure if this is overdoing it and whether it would make a difference anywhere.

Reply:
@davidrohr not really: in the final version the ``mRefTPC`` may refer also to TPC track constrained by TOF, TRD or both (GlobalTrackID::TPCTOF, GlobalTrackID::TPCTRD...). Similarly, for the TPCtrack-ITSclusters afterburner I'll need to introduce separate container with matched ITS cluster IDs, this will be yet another GlobalTrackID::Source entry which will be registered in the mRefITS.
For the clear-cut cases I indeed use integers, like the reference on the TOF cluster matched to global track:
https://github.com/AliceO2Group/AliceO2/blob/b5e69045cd2beac57f9ff844b7693470ee95ca60/DataFormats/Reconstruction/include/ReconstructionDataFormats/MatchInfoTOF.h#L51-L52
Though, in this particular case there is another problem: TOF still uses for indexing a ``EvIndex<int, int>`` to store {tree_entry_ID, index in entry}, so, I've just substituted the reference on the track by ``EvIndex<int, GlobalTrackID>``.
This is a legacy from the times when we did not have just 1 entry per TF (we still have multiple entries for TPC and ITS "triggered" modes, but these trigger implementations are anyway wrong, I would simply introduce extra vector delimiting the ranges of every trigger in a single tree entry). @noferini, I would suppress this tree entry references.

Reply:
ok, got it. Initially I was thinking that these constrained tracks of course also have a TPC track ID, but obviously you need the updated constrained track parameters... Makes totally sense as is.